aisuite/providers/anthropic_provider.py

```diff
@@ -15,26 +34,195 @@ def __init__(self, **config):
 
         self.client = anthropic.Anthropic(**config)
 
+    def convert_request(self, messages):
+        """Convert framework messages to Anthropic format."""
+        return [self._convert_single_message(msg) for msg in messages]
+
+    def _convert_single_message(self, msg):
+        """Convert a single message to Anthropic format."""
+        if isinstance(msg, dict):
+            return self._convert_dict_message(msg)
+        return self._convert_message_object(msg)
```

Comment:
It looks like both `_convert_dict_message` and `_convert_message_object` do the same things. Would it be better to convert to one format and keep one of the convert message methods.

```python
if isinstance(msg, dict):
    msg = Message(msg)
return self._convert_message_object(msg)
```

or the other way
```python
if isinstance(msg, Message):
    msg = dict(msg)
return self._convert_dict_message(msg)
```